enhancements/network/ovs-hardware-offload.md

```diff
@@ -0,0 +1,242 @@
+created and used as geneve endpoint of OVS bridge. With the SR-IOV VF LAG
+capability, the NIC PFs can receive the rules that the OVS tries to offload to
+the linux bond net device and offload them to the hardware e-switch.
+
+#### RHCOS
+
+- RHEL 8.3 be used as RHCOS image base for fully support of OVS Hardware Offload
+with connection track feature.
+
+#### OpenvSwitch
+
+- OpenvSwitch package [version 2.13](https://github.com/openshift/ovn-kubernetes/pull/122) is currently used in OpenShift.
+
+#### OVN-Kubernetes
+
+- OVS Hardware Offload is a [feature available](https://github.com/ovn-org/ovn-kubernetes/blob/master/docs/ovs_offload.md) in OVN-Kubernetes upstream.
+
+- `ovn-k8s-cni-overlay` CNI plugin configures the pod interface by moving the VF
+to pod namespace and plugging VF representor to `br-int` OVS bridge. The
+relevant [patches](https://github.com/openshift/ovn-kubernetes/commit/6c96467d0d3e58cab05641293d1c1b75e5914795) are available downstream.
+
+#### Cluster Network Operator
+
+- Cluster Network Operator will enable hardware offload and configure tc-policy
+on node OpenvSwitch instance and restart OpenvSwitch instance for hardware
+offload change to take effect.
+
+- Enabling `hw-offload` in CNO `ovs-node` DaemonSet is equivalent to executing
+`ovs-vsctl set Open_vSwitch . other_config:hw-offload=true`. The default value
+is `false`.
+
+- Setting `tc-policy` in CNO `ovs-node` DaemonSet is equivalent to executing
+`ovs-vsctl set Open_vSwitch . other_config:tc-policy=skip_sw`. The default
+value is `none`, optional values can be one of `none`, `skip_sw` and `skip_hw`.
+This field is only relevant if `hw-offload` is enabled.
+
+#### Multus
+
+- Multus will support specifying net-attach-def with customized annotations for
+[default cluster network in pod annotations](https://github.com/intel/multus-cni/blob/master/doc/configuration.md#specify-default-cluster-network-in-pod-annotations).
+Customized annotations will contain a resource name requesting SR-IOV device.
+
+#### SR-IOV Operator
+
+- SR-IOV Operator will support new device type `switchdev` in Operator
+[SriovNetworkNodePolicy API](https://github.com/openshift/sriov-network-operator/blob/master/pkg/apis/sriovnetwork/v1/sriovnetworknodepolicy_types.go#L33).
+
+- SR-IOV config daemon (a DaemonSet managed by SR-IOV Operator) will support
+changing e-switch mode from `legacy` to `switchdev` on the PF device and
+exposing VF representor information in `SriovNetworkNodeState` status.
+
+- SR-IOV network device plugin (a Device Plugin DaemonSet managed by SR-IOV
+Operator) supports advertising VF resource to kubelet.
+
+#### Machine Config Operator (MCO) [optional]
+
+- MCO will be used to create linux bond interface on two identical PFs that
+are capable of doing OVS hardware offload and create vlan interfaces on bond
+interface. The supported bond modes are `Active-Backup`, `Active-Active`
+and `LACP`. This is not required if OVS Hardware Offload is used on PF
+device directly.
+
```

Comment:
How about enabling `iommu` in the kernel args?

Reply:
This can be done via SR-IOV Operator by adding a default value for iommu when `switchdev` type is specified. It's also possible to use MCO for changing kernel argument.